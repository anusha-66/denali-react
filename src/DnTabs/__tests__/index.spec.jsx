/**
 * Copyright 2020, Verizon Media
 * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
 */

import React from 'react';
import { render } from 'enzyme';
import { DnTabs, DnTabsStyles } from '../../index'; // src/index.tsx

describe('DnTabs Tests', () => {
    it('should export DnTabsStyles', () => {
        expect.assertions(1);

        expect(DnTabsStyles).toStrictEqual({
            primary: 'primary',
            secondary: 'secondary',
        });
    });

    it('should render with default props', () => {
        expect.assertions(1);

        const rendered = render(
            <DnTabs>
                <DnTabs.Tab>Tab 1</DnTabs.Tab>
                <DnTabs.Tab>Tab 2</DnTabs.Tab>
                <DnTabs.Tab>Tab 3</DnTabs.Tab>
                <DnTabs.Tab>Tab 4</DnTabs.Tab>
            </DnTabs>,
        );

        expect(rendered).toMatchSnapshot();
    });

    it('should render correctly with custom props', () => {
        expect.assertions(1);

        const rendered = render(
            <DnTabs isVertical style={DnTabsStyles.secondary} className="dummy-tabs">
                <DnTabs.Tab className="dummy-tab-item">Tab 1</DnTabs.Tab>
                <DnTabs.Tab isActive>Tab 2</DnTabs.Tab>
                <DnTabs.Tab isDisabled>Tab 3</DnTabs.Tab>
                <DnTabs.Tab>Tab 4</DnTabs.Tab>
            </DnTabs>,
        );

        expect(rendered).toMatchSnapshot();
    });

    it('should render correctly with section header', () => {
        expect.assertions(1);

        const rendered = render(
            <DnTabs isTabletDownHideLeft isVertical style={DnTabsStyles.secondary} className="dummy-tabs">
                <DnTabs.SectionHeader className="dummy-section-header-item">SECTION HEADER</DnTabs.SectionHeader>
                <DnTabs.Tab isActive>Tab 1</DnTabs.Tab>
                <DnTabs.Tab isDisabled>Tab 2</DnTabs.Tab>
                <DnTabs.Tab>Tab 3</DnTabs.Tab>
            </DnTabs>,
        );

        expect(rendered).toMatchSnapshot();
    });
});
