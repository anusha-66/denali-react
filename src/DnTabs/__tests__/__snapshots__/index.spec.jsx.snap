// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DnTabs Tests should render correctly with custom props 1`] = `
<div
  class="tabs is-secondary is-vertical dummy-tabs"
>
  <ul>
    <li
      class="dummy-tab-item"
    >
      <span>
        Tab 1
      </span>
    </li>
    <li
      class="is-active"
    >
      <span>
        Tab 2
      </span>
    </li>
    <li
      class="is-disabled"
    >
      <span>
        Tab 3
      </span>
    </li>
    <li
      class=""
    >
      <span>
        Tab 4
      </span>
    </li>
  </ul>
</div>
`;

exports[`DnTabs Tests should render correctly with section header 1`] = `
<div
  class="tabs is-secondary is-vertical tablet-down-hide-left dummy-tabs"
>
  <ul>
    <li
      class="tabs__section-header dummy-section-header-item"
    >
      <span>
        SECTION HEADER
      </span>
    </li>
    <li
      class="is-active"
    >
      <span>
        Tab 1
      </span>
    </li>
    <li
      class="is-disabled"
    >
      <span>
        Tab 2
      </span>
    </li>
    <li
      class=""
    >
      <span>
        Tab 3
      </span>
    </li>
  </ul>
</div>
`;

exports[`DnTabs Tests should render with default props 1`] = `
<div
  class="tabs is-primary"
>
  <ul>
    <li
      class=""
    >
      <span>
        Tab 1
      </span>
    </li>
    <li
      class=""
    >
      <span>
        Tab 2
      </span>
    </li>
    <li
      class=""
    >
      <span>
        Tab 3
      </span>
    </li>
    <li
      class=""
    >
      <span>
        Tab 4
      </span>
    </li>
  </ul>
</div>
`;
